### ISSUE TITLE
Android: Hints not announced on some components

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30847]
## Description

- [ ] #31056 Hint set on an element without an accessibilityLabel, does *not* pull in descendant text to use as its label but instead announces the hint only. 

- [ ] #31057 Hint not announced on Button or Picker (This is likely related to #30842 and #30843 , as these elements also do not properly support other accessibility attributes).  

## React Native version:
v0.63

## Snack
https://snack.expo.io/JDrZlRFtP

## Expected Behavior
Upon focus of an element with a hint and no accessibilityLabel, the element's contents would be announced, followed by the hint.  On all component types this should work the same way.

## Android Details
This is occurring due to the fact that Android does not have a generic "hint" property like iOS does. So to fake this behavior React Native simply appends the "hint" to Android's contentDescription property. The problem is that if an element has no contentDescription defined, we are appending the hint alone, which sets the contentDesceiption to just the hint, overriding any element text content, which would normally be announced if no accessibilityLabel was defined.

To fix this we would need to derive what the announced text *would be* without the hint, and set that as the contentDescription before appending the hint OR use a different property, such as toolTipText instead.

Knowing the exact announcement is a difficult endeavor, because Android will combine the text of an element and any non-focusable descendants to determine what it announces. This means we would need to traverse the entire tree downwards from this element to determine what its label might be. This would add a significant performance cost, as well as significant complexity (and room for bugs), and would rely on complex Talkback behavior that might later change. Because of this, I'd recommend that we do not do this, and instead use the AccessibilityNodeInfo's "tooltipText" property to announce the hint.

TooltipText is meant to be used for visible Tooltips associated with a View, but has the added benefit that when it is added *only* on the AccessibilityNodeInfo (and not the View), it will simply announce whatever string is passed to it after the main focus announcement.  This is as close an analog to iOS's generic "hint" feature as exists on Android.

**Comment by blavalla** [comment]
@amarlette, we may want to split the <Button> and <Picker> issue out into its own task, as it's a far simpler fix than the quite complicated issue of the hint and element text not being properly combined.  That one would make a good first task, but the other issue would not.

**Comment by amarlette** [comment]
Ok I can split this into two tasks and link them here

**Comment by blavalla** [comment]
This parent issue is no longer useful for tracking, see sub-issues for details.
